types/src/v17/control.rs

```diff
@@ -52,25 +52,34 @@ pub struct Locked {
 /// > Gets and sets the logging configuration.
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub struct Logging {
-    pub net: bool,
-    pub tor: bool,
-    pub mempool: bool,
-    pub http: bool,
-    pub bench: bool,
-    pub zmq: bool,
-    pub db: bool,
-    pub rpc: bool,
-    pub estimatefee: bool,
     pub addrman: bool,
-    pub selectcoins: bool,
-    pub reindex: bool,
+    pub bench: bool,
+    pub blockstorage: Option<bool>, // v23 and later only
```

Comment:
TL;DR At this stage I don't want to make changes like this because it invalidates the whole design of the crate. 

I accidentally did this same thing once but I came to the conclusion that doing so totally invalidates the whole design of the crate. The _whole point_ of `v17::Foo` is that one knows that this is the exact shape of a `Foo` returned by Bitcoin Core v17. If one wants to not think about the version and check all the optional fields then one can first convert to a `model`  type and do so. The central claim of this repo is that these use cases are different and keeping them separate will lead to better software. This was an intuitive leap and only time will tell if I was correct.